src/lib/middleware/bm_service_request.cpp

```diff
@@ -141,7 +151,9 @@ static bool _request_list_remove_request(bm_service_request_node_t * node) {
 static bm_service_request_node_t * _create_node(size_t service_strlen, const char * service, bm_service_reply_cb reply_cb, uint32_t timeout_ms) {
     bm_service_request_node_t * node = static_cast<bm_service_request_node_t*>(pvPortMalloc(sizeof(bm_service_request_node_t)));
     configASSERT(node);
-    node->service = service;
+    node->service = static_cast<char*>(pvPortMalloc(service_strlen));
```

Comment:
Adding support so our nodes always have reference to their service name (in case the caller frees the name buffer)